### ISSUE TITLE
Allow anonymous users to export search results to CSV

### ISSUE THREAD

**Issue by alanorth** [DSpace/dspace-angular#3258]
## Is your feature request related to a problem? Please describe.
In DSpace 6 it was possible for users to export search results to CSV. As of DSpace 7 this is only possible for administrators, perhaps because of limitations in the way Angular interacts with the REST API to create processes.

## Describe the solution you'd like
Allowing users to export search results as CSV is important for librarians, researchers, and more. Furthermore, it is important for reproducibility, for example in the case of systematic literature reviews where you must specify the search terms used when registering the protocol.

## Describe alternatives or workarounds you've considered
A workaround for advanced users familiar with [DSpace advanced search syntax](https://wiki.lyrasis.org/display/DSDOC7x/Search+-+Advanced) (and boolean search in general) would be to use the REST API directly to retrieve results in JSON.

For example, an advanced search and equivalent REST API search that both return the same number of results:

- Advanced search with facet filters: https://demo.dspace.org/search?query=&f.entityType=Publication%2Cequals&spc.page=1&f.dateIssued.min=2012&f.dateIssued.max=2023
- Equivalent REST API search: https://demo.dspace.org/server/api/discover/search/objects?query=dspace.entity.type%3APublication%20AND%20dc.date.issued%3A%5B2012%20TO%202023%5D

The user must not only be familiar with advanced search syntax, but also the DSpace REST API, JSON, metadata fields, URL encoding, and more.

## Additional information

- There is a concern that CSV export can produce a large number of results and cause the server to hang — https://github.com/DSpace/DSpace/issues/9399

**Comment by tdonohue** [comment]
I agree this would be a nice feature.  However, it would rely on a way to allow anonymous uses access to **Processes** in the User Interface (as this export is triggered via a Process).  This may be complex to achieve unless we figure out a different way for the export to occur for anonymous users.

Another option is to consider allowing this feature for *registered users only* and then finding a way to allow all logged in users to access **Processes**.

Overall, I like the idea.  I'm just pointing out that this is very complex to achieve using the current Processes approach.

**Comment by arvoConsultores** [comment]
Hi, we have developed a variation (for Dspace 7.6) for logged-in users. 

Description follows: 
1) User Type Check:
Implemented a method to check if the user is an administrator or logged-in. This determines whether the search results export button is displayed or not.
Administrators have full access, while logged-in users can download only the exportmetadata processes.
2) Backend Processes:
When starting the export process, the user type is checked.
If the user is an administrator, he has all privileges. If he is logged-in, he can only access the metadata-export-search script.

3) Metadata Download Configuration:
In the file that configures the download of metadata records, it is checked if the user is an administrator.
If he is an administrator, all records are exported; if not, the export is limited to 500 records (configurable)

4) Frontend Changes:
Modified  the permissions in the frontend so that registered users can see the export button, which was previously reserved only for administrators.
